protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -934,3 +911,27 @@ class _FieldSet {
     _oneofCases?.addAll(original._oneofCases!);
   }
 }
+
+// Hashes the value of one field (recursively).
+int _hashField(int hash, FieldInfo fi, value) {
```

Comment:
For readability, I would put this at line 689 immediately after the caller, and use the `static` keyword.

Reply:
Done.